src/backend/distributed/utils/reference_table_utils.c

```diff
@@ -76,6 +80,15 @@ upgrade_to_reference_table(PG_FUNCTION_ARGS)
 								  relationName)));
 	}
 
+	if (tableEntry->replicationModel == REPLICATION_MODEL_STREAMING)
+	{
+		char *relationName = get_rel_name(relationId);
+		ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
```

Comment:
for the sake of symmetry, I suggest leaving it this way since all other error messages are `cannot upgrade to reference table`.  For error detail, your suggestion looks good to me.